### PR TITLE
Feature/change password

### PR DESCRIPTION
**Pull request by myke-awoniran**
finished Change Password Endpoint for an Already loggedIn user
Complete Endpoint Yaml specification

### CODE REVIEW

File: spec/api.yaml [hunk 1]
```diff
@@ -358,6 +390,18 @@ components:
...
+    ChangeUserPassword:
+      type: object
+      required:
+        - currentPassword
+        - newPassword
+      properties:
+        currentPassword:
+          type: string
+        newPassword:
+          type: string
```
Comment (Lekky71) [inline spec/api.yaml:403]: No validation. Kindly add, you can refer to the SignUpRequest.

File: src/controlllers/user.controller.ts [hunk 2]
```diff
@@ -12,3 +12,17 @@ export async function handleGetMyProfile(req: IExpressRequest, res: ExpressRespo
     ResponseManager.handleError(res, err);
   }
 }
+
+export async function handleManagePassword(req: IExpressRequest, res: ExpressResponse): Promise<void> {
```
Comment (Lekky71) [inline src/controlllers/user.controller.ts]: `handleChangePassword` ?

File: src/controlllers/user.controller.ts [hunk 3]
```diff
@@ -12,3 +12,17 @@ export async function handleGetMyProfile(req: IExpressRequest, res: ExpressRespo
...
+  try {
+    const body = {
+      userId: <string>req.userId,
+      currentPassword: req.body.currentPassword,
+      newPassword: req.body.newPassword
+    };
+    const response = await userService.managePassword(body)
+    ResponseManager.success(res, response)
```
Comment (Lekky71) [inline src/controlllers/user.controller.ts]: End statements with semicolon. ;

File: src/services/user.service.ts [hunk 4]
```diff
@@ -8,3 +8,34 @@ export async function getMyProfile(userId: string): Promise<User> {
...
+  /**
+   * check if there is a user
+   * verify current password
+   * if the current password is incorrect, break code execution
+   * true? set new password,
+   * delete all token for the user
+   * return user
+   * */
+
+  if (!userAuth.verifyPassword(props.currentPassword))
```
Comment (Lekky71) [inline src/services/user.service.ts]: Always put braces around IF statements.

**Review by Lekky71 — CHANGES_REQUESTED** [review]
Good work just some changes to be made.

File: src/controlllers/user.controller.ts (hunk 2 above)
Comment (myke-awoniran) [reply src/controlllers/user.controller.ts]: yes

**Review by Lekky71 — APPROVED** [review]

**Pull request merged**